controller/k8s/api.go

```diff
@@ -273,22 +274,34 @@ func (api *API) GetPodsFor(obj runtime.Object, includeFailed bool) ([]*apiv1.Pod
 	return allPods, nil
 }
 
+// getNamespaces returns the namespace matching the specified name. If no name
+// is given, it returns all namespaces, unless the API was configured to only
+// work with a single namespace, in which case it returns that namespace. Note
+// that namespace reads are not cached.
 func (api *API) getNamespaces(name string) ([]runtime.Object, error) {
-	var err error
-	var namespaces []*apiv1.Namespace
+	namespaces := make([]*apiv1.Namespace, 0)
+
+	if name == "" && api.namespace != "" {
```

Comment:
if `name != "" && name != api.namespace`, should we log and/or return an error?

Reply:
If we return an error here, that would break the CLI and web frontend, which omit the namespace when requesting stat/tap data for all namespaces. Instead, in single namespace mode we want to pretend that there's only one namespace in the cluster.